oxidation/libparsec/crates/crypto/src/sodiumoxide/sequester.rs

```diff
@@ -241,8 +241,14 @@ impl Serialize for SequesterPublicKeyDer {
 
 impl PartialEq for SequesterPublicKeyDer {
     fn eq(&self, other: &Self) -> bool {
-        let public_key = PKey::from_rsa(self.0.to_owned()).expect("Unreachable");
-        public_key.public_eq(&PKey::from_rsa(other.0.to_owned()).expect("Unreachable"))
+        // This is the only way I could find to compare two public keys.
+        // There is also public_eq, but it is defined on the PKey type, and to get
+        // a PKey<Public> from a Rsa<Public> we have to give up ownership, which
+        // we can't do because we take a reference to both the keys we want to compare.
+        // We could also clone the keys, but that would mean having an allocation in the
+        // comparison method.
+        self.0.public_key_to_der().expect("OpenSSL error")
```

Comment:
`public_key_to_der` result in a Vec<u8>, so you allocate :/
Why don't use his component `n` and `e` ?

Reply:
I was unsure that having the same modulus and exponent guaranteed the key was the same (I tried to google but I was missing the appropriate keywords).

I will update right now.